.github/workflows/changelog.yml

```diff
@@ -15,19 +16,87 @@ jobs:
           git fetch origin ${{ github.base_ref }} --depth=1
           if [[ $(git diff --name-only FETCH_HEAD | grep CHANGELOG) ]]
           then
-            echo "A CHANGELOG was modified. Looks good!"
+            echo "The CHANGELOG should not be directly modified."
+            echo "Please add a .yaml file to the ./unreleased/ directory."
+            echo "See CONTRIBUTING.md for more details."
+            echo "Alternately, add the \"Skip Changelog\" label if this job should be skipped."
+            false
+          else
+            echo "The CHANGELOG was not modified."
+          fi
+
+      - name: Ensure one ./unreleased/*.yaml addition
+        run: |
+          git fetch origin ${{ github.base_ref }} --depth=1
+          if [[ 1 -gt $(git diff --diff-filter=A --name-only FETCH_HEAD ./unreleased | grep -c \\.yaml) ]]
+          then
+            echo "No changelog entry was added to the ./unreleased/ directory."
+            echo "Please add a .yaml file to the ./unreleased/ directory."
+            echo "See CONTRIBUTING.md for more details."
+            echo "Alternately, add the \"Skip Changelog\" label if this job should be skipped."
+            false
+          elif [[ 1 -lt $(git diff --diff-filter=A --name-only FETCH_HEAD ./unreleased | grep -c \\.yaml) ]]
+          then
+            echo "More than 1 changelog entry was added to the ./unreleased/ directory."
+            echo "Please add a .yaml file to the ./unreleased/ directory."
+            echo "See CONTRIBUTING.md for more details."
+            echo "Alternately, add the \"Skip Changelog\" label if this job should be skipped."
+            false
           else
-            echo "No CHANGELOG was modified."
-            echo "Please add a CHANGELOG entry, or add the \"Skip Changelog\" label if not required."
+            echo "A changelog entry was added to the ./unreleased/ directory!"
+          fi
+
+      - name: Ensure no other ./unreleased/*.yaml changes
+        run: |
+          git fetch origin ${{ github.base_ref }} --depth=1
+          if [[ 0 -eq $(git diff --diff-filter=CDMRTUXB --name-only FETCH_HEAD ./unreleased | grep -c \\.yaml) ]]
+          then
+            echo "No other changes to ./unreleased/ directory!"
+          else
+            echo "Unrelated changes to ./unreleased/ directory detected."
+            echo "Only a single file may be added to ./unreleased/ directory per PR. No other changes are allowed to this directory."
+            echo "See CONTRIBUTING.md for more details."
+            echo "Alternately, add the \"Skip Changelog\" label if this job should be skipped."
             false
           fi
+
+      - name: Validate ./unreleased/*.yaml changes
+        run: |
+          make chlog-validate \
+            || { echo "New ./unreleased/*.yaml file failed validation."; exit 1; }
+
+      # In order to validate any links in the yaml file, render the config to markdown
+      - name: Render unreleased changelog entries
```

Comment:
I don't think it's possible yet. We have a few places where reuse would be helpful. Agree with the notion though. I considered putting this check in the `check-link` workflow, but then we have to install go over there, so it seemed both options were less than ideal.

Reply:
Ya an action would probably be the best way to reuse this logic instead of chaining workflows together.  May not be worth the effort, the logic really shouldn't change that often